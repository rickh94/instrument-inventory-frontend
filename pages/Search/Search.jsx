import React, { useContext } from 'react'
import PropTypes from 'prop-types'

import { RootPaper, SearchResultsList, FindInstrument } from '../../components'
import { HelpersContext } from '../../contexts'

const Search = ({}) => {
  const { showAlert, searchResults, setSearchResults } = useContext(HelpersContext)

  return (
    <RootPaper>
      <FindInstrument showMultipleResults={setSearchResults} showAlert={showAlert} />
      {searchResults.length > 0 && <SearchResultsList results={searchResults} />}
    </RootPaper>
  )
}

Search.propTypes = {
}

export default Search
