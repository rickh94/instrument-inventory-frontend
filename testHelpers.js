/* eslint-disable react/prop-types */
import React from 'react'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core'
import { HelpersContext, SchemaContext } from './contexts'
import { MemoryRouter } from 'react-router-dom'

export const TestTheme = ({ children }) => (
  <MuiThemeProvider theme={createMuiTheme()}>{children}</MuiThemeProvider>
)


export const TestHelpers = ({
  children,
  showAlert = jest.fn(),
  setSearchResults = jest.fn(),
  setFilterResults = jest.fn(),
  searchResults = [],
  filterResults = [],
}) => (
  <HelpersContext.Provider
    value={{
      showAlert,
      setSearchResults,
      setFilterResults,
      searchResults,
      filterResults,
    }}
  >
    {children}
  </HelpersContext.Provider>
)

export const TestRouter = ({ children }) => <MemoryRouter>{children}</MemoryRouter>

export const TestSchema = ({ children, schema = { components: { schemas: {} } } }) => (
  <SchemaContext.Provider value={schema}>{children}</SchemaContext.Provider>
)

export const TestEverything = ({ children, schema, helpers }) => (
  <TestTheme>
    <TestHelpers {...helpers}>
      <TestSchema schema={schema}>{children}</TestSchema>
    </TestHelpers>
  </TestTheme>
)

export const flushPromises = () => new Promise(setImmediate)
