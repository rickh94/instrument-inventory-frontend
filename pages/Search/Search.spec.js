import React from 'react'
import Search from './Search'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { TestEverything } from '../../testHelpers'

describe('<Search />', () => {
  it('matches snapshot', () => {
    const { container } = render(<Search />, { wrapper: TestEverything })
    expect(container).toMatchSnapshot()
  })
})
