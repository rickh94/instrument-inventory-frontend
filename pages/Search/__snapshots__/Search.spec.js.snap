// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded WithStyles(ForwardRef(Paper))-root-1"
  >
    <div
      class="makeStyles-heading-30"
      data-testid="loading-header"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
      >
        Find an Instrument
      </h5>
    </div>
    <form>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          data-testid="label"
          for="scanner-field"
        >
          Name or Number
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          data-testid="input-field"
          required=""
        >
          <input
            aria-describedby="scanner-error"
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
            id="scanner-field"
            required=""
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              data-testid="scan-button"
              tabindex="0"
              title="Scan Barcode"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-barcode fa-w-16 "
                  data-icon="barcode"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 448V64h18v384H0zm26.857-.273V64H36v383.727h-9.143zm27.143 0V64h8.857v383.727H54zm44.857 0V64h8.857v383.727h-8.857zm36 0V64h17.714v383.727h-17.714zm44.857 0V64h8.857v383.727h-8.857zm18 0V64h8.857v383.727h-8.857zm18 0V64h8.857v383.727h-8.857zm35.715 0V64h18v383.727h-18zm44.857 0V64h18v383.727h-18zm35.999 0V64h18.001v383.727h-18.001zm36.001 0V64h18.001v383.727h-18.001zm26.857 0V64h18v383.727h-18zm45.143 0V64h26.857v383.727h-26.857zm35.714 0V64h9.143v383.727H476zm18 .273V64h18v384h-18z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="MuiTypography-root MuiTypography-srOnly"
                  data-testid="sr-text"
                >
                  Scan Barcode
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        
      </div>
      <div
        class="MuiFormGroup-root MuiFormGroup-row"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-lastButton-29 MuiButton-text"
          data-testid="clear-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Clear
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-textPrimary MuiButton-text Mui-disabled Mui-disabled"
          data-testid="submit-button"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
